### PR TITLE
Update API to read from Zarr arrays

### PR DESCRIPTION
**Pull request by esheehan-gsl**
Modify the existing API to read from Zarr arrays that match those produced by the data pipeline service (#67). The API itself remains unchanged.

This PR is stacked on top of #135.

### CODE REVIEW

File: services/api/src/unified_graphics/diag.py [hunk 1]
```diff
@@ -69,120 +68,63 @@ def to_geojson(self):
...
+
+    return S3Map(root=f"{result.netloc}{result.path}", s3=s3, check=False)
+
+
+def open_diagnostic(
+    variable: Variable, initialization_time: str, loop: MinimLoop
+) -> xr.Dataset:
+    store = get_store(current_app.config["DIAG_ZARR"])
+    group = f"/{variable.value}/{initialization_time}/{loop.value}"
+    return xr.open_zarr(store, group=group)
```
Comment (ian-noaa) [inline services/api/src/unified_graphics/diag.py:95]: 👏 Using `urlparse` & `s3fs` here in place of `fsspec` really does remove a good amount of complexity. Nicely done.

File: services/api/tests/conftest.py [hunk 2]
```diff
@@ -1,24 +1,41 @@
...
+    if config.getoption("--runaws"):
+        # If --runaws is set, there's no need to modify any of the tests
+        # to mark them skipped.
+        return
+
+    # Mark any test with the aws mark as skipped because --runaws is off
+    skip_aws = pytest.mark.skip(reason="use --runaws to run")
+    for item in items:
+        if "aws" in item.keywords:
+            item.add_marker(skip_aws)
```
Comment (ian-noaa) [inline services/api/tests/conftest.py:38]: This is a nice way to automatically disable tests.

**Review by ian-noaa — APPROVED** [review]
Directing xarray to use the Zarr backend instead of having it guess really removed a lot of complexity from the code. Moving away from Moto definitely has some downsides but it's certainly simplified the test suite - that fixture setting up moto wasn't the prettiest. I think the `--runaws` option you added will work well for making sure we don't run against AWS when we don't want to or can't.

This looks great to me.

File: services/api/src/unified_graphics/diag.py (hunk 1 above)
Comment (esheehan-gsl) [reply services/api/src/unified_graphics/diag.py:95]: Yeah, it’s one of the advantages of Zarr. Pretty sure this wouldn’t work with NetCDF, although I’ve never tried.

File: services/api/tests/conftest.py (hunk 2 above)
Comment (esheehan-gsl) [reply services/api/tests/conftest.py:38]: Straight out of the pytest docs. I always forget where it is, but I know it’s in there, so I just have to keep digging until I find it.

**Comment by esheehan-gsl** [comment]
@ian-noaa When we deploy this, we’ll need to update the runtime environment to set `FLASK_DIAG_ZARR` to be `s3://osti-modeling-dev-rtma-vis/diag.zarr/` instead of the current `FLASK_DIAG_DIR`.

**Comment by ian-noaa** [comment]
Sounds good. I'll have VLab make those changes.

I had VLab set up a new S3 bucket for the lambda function to drop the "production" Zarr files into. Once the lambda is up and running, we should be able to switch the `api` service to use: `s3://osti-modeling-dev-rtma-vis-prod/diagnostics.zarr`.

**Pull request merged**